src/main/java/software/amazon/event/kafkaconnector/mapping/DefaultEventBridgeMapper.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.event.kafkaconnector.mapping;
+
+import static java.util.Collections.singletonMap;
+import static java.util.stream.Collectors.partitioningBy;
+import static java.util.stream.Collectors.toList;
+import static software.amazon.event.kafkaconnector.EventBridgeResult.failure;
+import static software.amazon.event.kafkaconnector.EventBridgeResult.success;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.apache.kafka.connect.header.Header;
+import org.apache.kafka.connect.json.JsonConverter;
+import org.apache.kafka.connect.sink.SinkRecord;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.event.kafkaconnector.EventBridgeResult;
+import software.amazon.event.kafkaconnector.EventBridgeSinkConfig;
+
+public class DefaultEventBridgeMapper implements EventBridgeMapper {
+
+  private static final String sourcePrefix = "kafka-connect.";
+
+  private final EventBridgeSinkConfig config;
+  private final JsonConverter jsonConverter = new JsonConverter();
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  public DefaultEventBridgeMapper(EventBridgeSinkConfig config) {
+    jsonConverter.configure(singletonMap("schemas.enable", "false"), false);
+    this.config = config;
+  }
+
+  public EventBridgeMappingResult map(List<SinkRecord> records) {
+    var partition =
+        records.stream()
+            .map(this::createPutEventsEntry)
+            .collect(partitioningBy(EventBridgeResult::isSuccess));
+
+    var success = partition.get(true).stream().map(EventBridgeResult::success).collect(toList());
+    var errors = partition.get(false).stream().map(EventBridgeResult::failure).collect(toList());
+
+    return new EventBridgeMappingResult(success, errors);
+  }
+
+  private EventBridgeResult<PutEventsRequestEntry> createPutEventsEntry(SinkRecord record) {
+    try {
+      return success(
+          record,
+          PutEventsRequestEntry.builder()
+              .eventBusName(config.eventBusArn)
+              .source(sourcePrefix + config.connectorId)
+              .detailType(config.getDetailType(record.topic()))
+              .resources(config.resources)
+              .detail(createJsonPayload(record))
+              .build());
+    } catch (JsonProcessingException e) {
```

Comment:
Question: any other exceptions of interest here to avoid panicing?

Reply:
Propagated exceptions from `createJsonPayload`, but to be safe we can catch `Exception` if `PutEventsRequestEntry.builder()....build()` throws any.